packages/authorization-server/src/routers/authorizationRouter.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  initFileManager,
+  initRedisRateLimiter,
+  InteropTokenGenerator,
+  logger,
+  LoggerMetadata,
+  rateLimiterHeadersFromStatus,
+} from "pagopa-interop-commons";
+import { fastifyFormbody } from "@fastify/formbody";
+import Fastify, { FastifyRequest } from "fastify";
+import { FastifyInstance } from "fastify";
+import {
+  CorrelationId,
+  generateId,
+  tooManyRequestsError,
+} from "pagopa-interop-models";
+import { authorizationServerApi } from "pagopa-interop-api-clients";
+import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
+import { initProducer } from "kafka-iam-auth";
+import { makeApiProblem } from ".././model/domain/errors.js";
+import { authorizationServerErrorMapper } from ".././utilities/errorMappers.js";
+import { tokenServiceBuilder } from ".././services/tokenService.js";
+import { config } from ".././config/config.js";
+import { InteropTokenResponse } from "../model/domain/models.js";
+
+const serviceName = "authorization-server";
+
+const dynamoDBClient = new DynamoDBClient({});
+const redisRateLimiter = await initRedisRateLimiter({
+  // TODO add limiter group
+  limiterGroup: "TODO",
+  maxRequests: config.rateLimiterMaxRequests,
+  rateInterval: config.rateLimiterRateInterval,
+  burstPercentage: config.rateLimiterBurstPercentage,
+  redisHost: config.rateLimiterRedisHost,
+  redisPort: config.rateLimiterRedisPort,
+  timeout: config.rateLimiterTimeout,
+});
+const producer = await initProducer(config, config.tokenAuditingTopic);
+const fileManager = initFileManager(config);
+
+const fastifyServer: FastifyInstance = Fastify({ logger: { level: "error" } });
+await fastifyServer.register(fastifyFormbody);
+
+const tokenGenerator = new InteropTokenGenerator({
+  generatedInteropTokenKid: config.generatedInteropTokenKid,
+  generatedInteropTokenIssuer: config.generatedInteropTokenIssuer,
+  generatedInteropTokenM2MAudience: config.generatedInteropTokenM2MAudience,
+  generatedInteropTokenM2MDurationSeconds:
+    config.generatedInteropTokenM2MDurationSeconds,
+});
+
+const tokenService = tokenServiceBuilder({
+  tokenGenerator,
+  dynamoDBClient,
+  redisRateLimiter,
+  producer,
+  fileManager,
+});
+
+fastifyServer.post(
+  "/token.oauth2",
+  async (
+    request: FastifyRequest<{
+      Body: authorizationServerApi.AccessTokenRequest;
+    }>,
+    reply
+  ) => {
+    const correlationId = generateId<CorrelationId>();
+    const loggerMetadata: LoggerMetadata = {
+      serviceName,
+      correlationId,
+    };
+    const loggerInstance = logger(loggerMetadata);
+
+    try {
+      const res = await tokenService.generateToken(
+        request.body,
+        correlationId,
+        loggerInstance
+      );
+
+      if (res.limitReached) {
+        const headers = rateLimiterHeadersFromStatus(res.rateLimiterStatus);
```

Comment:
I am just checking on rate limiter stuff because I saw the task is in review: in the other point where we use it manually, without the express middleware, we set the headers in any case, limit reached or success, so that we inform the caller about limit, interval, and remaining requests in advance

Reply:
Here's a link to the point I mention: https://vscode.dev/github.com/pagopa/interop-be-monorepo/blob/main/packages/backend-for-frontend/src/routers/authorizationRouter.ts#L49-L51

Reply:
Thanks, updated.